### ISSUE TITLE
thread 'main' panicked at 'index out of bounds: the len is 16 but the index is 16'

### ISSUE THREAD

**Issue by fourbytes** [dswd/vpncloud.rs#18]
Error - compiled with latest rust stable (rustc 1.17.0 (56124baa9 2017-04-24)):

```
root@server:~/vpncloud.rs# RUST_BACKTRACE=1 target/debug/vpncloud --config /etc/vpncloud/four.net 
INFO - Reading config file '/etc/vpncloud/four.net'
INFO - Opened device vpncloud0
WARN - Port-forwarding: no router found
thread 'main' panicked at 'index out of bounds: the len is 16 but the index is 16', /checkout/src/libcore/slice.rs:664
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:371
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:549
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:511
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:495
   7: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:471
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:69
   9: core::panicking::panic_bounds_check
             at /checkout/src/libcore/panicking.rs:56
  10: <usize as core::slice::SliceIndex<T>>::index_mut
             at /checkout/src/libcore/slice.rs:664
  11: core::slice::<impl core::ops::IndexMut<I> for [T]>::index_mut
             at /checkout/src/libcore/slice.rs:571
  12: <vpncloud::ip::RoutingTable as vpncloud::types::Table>::lookup
             at ./src/ip.rs:114
  13: <vpncloud::cloud::GenericCloud<P>>::handle_interface_data
             at ./src/cloud.rs:450
  14: <vpncloud::cloud::GenericCloud<P>>::run
             at ./src/cloud.rs:619
  15: vpncloud::run
             at ./src/main.rs:193
  16: vpncloud::main
             at ./src/main.rs:232
  17: std::panicking::try::do_call
             at /checkout/src/libstd/panicking.rs:454
  18: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
  19: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:433
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:57
  20: main
  21: __libc_start_main
  22: _start
```

Config:
```
device_type: tun
device_name: vpncloud%d
ifup: "ip link set dev $IFNAME mtu 1500 && ip link set $IFNAME up && ip addr add 10.42.1.22/16 dev $IFNAME"
crypto: aes256
shared_key: "SECRET_KEY"
port: 3210
peers:
 - serverA:3210
 - serverB:3210
 - serverC:3210
peer_timeout: 1800
mode: normal
subnets:
 - 10.42.1.22/32
```

**Comment by dswd** [comment]
I fixed the error in the last commit, bugfix release will follow

**Comment by dswd** [comment]
The bugfix is included in v0.8.1
